opa-manager-operator/files/sync.yaml.jinja2

```diff
@@ -0,0 +1,18 @@
+group: config.gatekeeper.sh
+version: v1alpha1
+plural: configs
+namespace: {{namespace}}
+body:
+  apiVersion: config.gatekeeper.sh/v1alpha1
+  kind: Config
```

Comment:
You should be able to set this through set_pod_spec as a custom resource. Does it not let you?

Reply:
IIRC there's a race condition here, but I will take another look and report back

Reply:
ah yes, there is a race condition here where the config.gatekeeper.sh is not yet created

Reply:
just adding detail to this:
```
Traceback (most recent call last):
  File "./src/charm.py", line 179, in <module>
    main(OPAManagerCharm)
  File "/var/lib/juju/agents/unit-gatekeeper-controller-manager-7/charm/venv/ops/main.py", line 406, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-gatekeeper-controller-manager-7/charm/venv/ops/main.py", line 140, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-gatekeeper-controller-manager-7/charm/venv/ops/framework.py", line 278, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-gatekeeper-controller-manager-7/charm/venv/ops/framework.py", line 722, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-gatekeeper-controller-manager-7/charm/venv/ops/framework.py", line 767, in _reemit
    custom_handler(event)
  File "./src/charm.py", line 57, in _on_config_changed
    self._configure_pod()
  File "./src/charm.py", line 174, in _configure_pod
    self.model.pod.set_spec(pod_spec)
  File "/var/lib/juju/agents/unit-gatekeeper-controller-manager-7/charm/venv/ops/model.py", line 955, in set_spec
    self._backend.pod_spec_set(spec, k8s_resources)
  File "/var/lib/juju/agents/unit-gatekeeper-controller-manager-7/charm/venv/ops/model.py", line 1415, in pod_spec_set
    self._run('pod-spec-set', *args)
  File "/var/lib/juju/agents/unit-gatekeeper-controller-manager-7/charm/venv/ops/model.py", line 1286, in _run
    raise ModelError(e.stderr)
ops.model.ModelError: b'ERROR json: unknown field "sync"\n'
application-gatekeeper-controller-manager: 14:12:55 ERROR juju.worker.caasoperator.uniter.gatekeeper-controller-manager/7.operation hook "config-changed" (via hook dispatching script: dispatch) failed: exit status 1
application-gatekeeper-controller-manager: 14:12:55 INFO juju.worker.caasoperator.uniter.gatekeeper-controller-manager/7 awaiting error resolution for "config-changed" hook
```

this is the issue I face when trying to deploy with the pod spec